### ISSUE TITLE
Satellite error (reading from server: read: connection timed out)

### ISSUE THREAD

**Issue by idelvall** [earthly/earthly#3255]
We have been experiencing this failure occasionally, when running in satellites over last days.
I've seen this happening both in GH actions as well as locally.
```
Init 🚀
————————————————————————————————————————————————————————————————————————————————

           satellite | satname is waking up. Please wait...
           satellite | ...amplifying transmission signal...
           satellite | ...testing antennas...
Shareable link: https://ci.earthly.dev/logs?logId=a9f1cc10-4427-4b05-b3ba-9244da7ffd25
Error: error reading from server: read tcp X.X.X.X:Y->100.21.239.443:443: read: connection timed out

Error: Process completed with exit code 1.
```

## Multiple jobs failing at the same time
The error is non-deterministic and happens rarely. 
The time window for reproducing it is narrow, since usually, when retrying it right away, it can't be reproduced, but if multiple GH jobs are running in parallel in the same satellite when it happens, the failure affects multiple of them.

For example, these two errors happened simultaneously in two different GH jobs. Notice the satellite state received was different, but still they failed:
```
Init 🚀
————————————————————————————————————————————————————————————————————————————————

           satellite | satname is waking up. Please wait...
           satellite | ...amplifying transmission signal...
           satellite | ...testing antennas...
Shareable link: https://ci.earthly.dev/logs?logId=a9f1cc10-4427-4b05-b3ba-9244da7ffd25
Error: error reading from server: read tcp X.X.X.X:Y->100.21.239.443:443: read: connection timed out

Error: Process completed with exit code 1.
```
```
Init 🚀
————————————————————————————————————————————————————————————————————————————————

           satellite | satname is starting. Please wait...
           satellite | ...perturbing matrices...
Shareable link: https://ci.earthly.dev/logs?logId=989e68d2-5577-4088-868a-649f0dec0852
Error: error reading from server: read tcp X.X.X.X:Y->100.21.239.443:443: read: connection timed out

Error: Process completed with exit code 1.
```

**Comment by brandonSc** [comment]
Adding a simple retry mechanism to the earthly client will help to recover from these situations, however, more investigation still needs to be done for us to understand why it is disconnecting in the first place.

**Comment by brandonSc** [comment]
> I am starting to think the problem the gRPC connection is terminated at that point, and the code times out later on when doing another call to the client.

I think it's a good theory. Although we usually see an error like `transport: Error while dialing dial tcp 10.4.10.26:8372: i/o timeout` when the next call (buildkit client connect) fails after reserve. The error message we get here is `read: connection timed out` makes me think it times out while reading from the stream.

**Comment by idelvall** [comment]
According to this https://www.evanjones.ca/tcp-connection-timeouts.html `read: connection timed out` happens when the timeout happens over an stablished connection. This seems aligned with my theory of the failure happening while reading the status stream, then that loop being gracefully exited, and then timing out in the next call to client.

`i/o timeout` seems to be associated to timing out while trying to establish a new connection

**Comment by idelvall** [comment]
Ok I was able to reproduce it. And turned out that my logic was right but my premise was wrong.

To reproduce it:
- sleep the satellite
- build on the satellite
- disconnect computer from the internet while showing the status updates


The error received was a `rpc error: code = Unavailable desc = error reading from server: read tcp 192.168.1.146:51556->100.22.9.17:443: read: connection timed out` on line https://github.com/earthly/earthly/blob/ffbc57da9490805667f2773207a7a827c00c8309/cloud/satellite.go#L252
 
The reason why we missed that was because for errors matching this regexp `(?U)rpc error: code = .+ desc = ` we were logging the root error rather than the wrapper https://github.com/earthly/earthly/blob/ffbc57da9490805667f2773207a7a827c00c8309/cmd/earthly/app/run.go#L265

Updating the PR to fix both the issue as well as the confusing logging logic

**Comment by brandonSc** [comment]
Good find @idelvall I didn't know we were stripping the RPC status code from the error. I think it could be useful information to keep for the user, but at the very least we should not discard all of the wrapped error messages.

**Comment by idelvall** [comment]
Fixed and released in `v0.7.19`

**Comment by fdietze** [comment]
I'm hitting this same problem every single time now. I just can't get it to pass.

https://github.com/social-protocols/GlobalBrain.jl/actions/runs/8898190228/job/24434788279?pr=5

A already pruned the cache and recreated the satellite.
